packages/terra-swap-provider/lib/TerraSwapProvider.ts

```diff
@@ -97,6 +98,10 @@ export default class TerraSwapProvider extends Provider implements Partial<SwapP
       throw new StandardError('Transactions are not matching')
     }
 
+    if (isTxError(initTx)) {
```

Comment:
Can we check for `isTxError` before `doesTransactionMatchInitiation`?

Reply:
done

Reply:
Do we want to make external calls/http calls before simple boolean checks?

Reply:
@harshjv wdyt and can we merge the code and update agent && wallet or there is something else which needs to be fixed

Reply:
Hello any updates on this @harshjv

Reply:
@kraikov `isTxError` is a synchronous method, so i do not think that its making any HTTP request.